utils/databake/derive/src/lib.rs

```diff
@@ -15,6 +15,22 @@ use syn::{
 };
 use synstructure::{AddBounds, Structure};
 
+/// This custom derive auto-implements the `Bake` trait on any given type that can be
+/// in theory const-constructed.
```

Comment:
I'd say we want people to try and use it mostly for const though, yes?

Reply:
How about I change this type to say something like "in most cases it is preferred if the output code works in const though we do not enforce that"

Reply:
People can use it however they want. The requirement for the derive is that all fields are public, nothing else. It will work in const if all contained values work in const, but I don't think we need to point that out. The crate docs have a section on const where you could add your preference, but because it's not an API requirement I don't see what the point is.

Note that I'm adding implementations for `Vec`, `String`, `Box` etc to databake in #2147

Reply:
Ah, hmmm. Gotcha.